apps/sms/services/js/messaging/messaging_service.js

```diff
@@ -0,0 +1,182 @@
+/* global bridge,
+          BridgeServiceMixin,
+          Settings,
+          SMIL
+*/
+'use strict';
+
+if (!('BridgeServiceMixin' in self)) {
+  importScripts('/services/js/bridge_service_mixin.js');
+}
+
+(function(exports) {
+
+const SERVICE_NAME = 'messaging-service';
+const MOBILE_MESSAGE_CLIENT_NAME = 'mozMobileMessageShim';
+
+const METHODS = Object.freeze([
+  'sendSMS', 'sendMMS', 'resendMessage', 'retrieveMMS'
+]);
+
+function ensureBridge() {
+  if (!('bridge' in self)) {
+    importScripts('/lib/bridge/bridge.js');
+  }
+}
+/**
+ * 0 is a valid value so we need to take care to not consider it as a falsy
+ * value. We want to return null for anything that's not a number or a
+ * string containing a number.
+ */
+function sanitizeServiceId(serviceId) {
+  if (serviceId == null || // null or undefined
+      isNaN(+serviceId)) {
+    serviceId = null;
+  } else {
+    serviceId = +serviceId;
+  }
+
+  return serviceId;
+}
+
+function getSendOptionsFromServiceId(serviceId) {
+  var sendOpts;
+
+  if (serviceId != null && // not null, not undefined
+      Settings.hasSeveralSim()) {
+    sendOpts = { serviceId: serviceId };
+  }
+
+  return sendOpts;
+}
+
+var MessagingService = {
+  /**
+   * Initializes our service using the Service mixin's initService and
+   * requests mozMobileMessage shim as client.
+   */
+  init() {
+    ensureBridge();
+
+    this.initService();
+
+    this.mozMobileMessageClient = bridge.client(MOBILE_MESSAGE_CLIENT_NAME);
+  },
+
+  /**
+   * Send sms by using send API shim.
+   * @param {String|String[]} recipients Contains the list of recipients for
+   *  this message.
+   * @param {String} content The message's body.
+   * @param {Number|String} serviceId The SIM serviceId we use to send the
+   *  message.
+   * @returns {Promise} Promise with array of the request result.
+   */
+  sendSMS({ recipients, content, serviceId }) {
+    recipients = recipients || [];
+    serviceId = sanitizeServiceId(serviceId);
+
+    if (!Array.isArray(recipients)) {
+      recipients = [recipients];
+    }
+
+    var sendOpts = getSendOptionsFromServiceId(serviceId);
+    var results = recipients.map(
+      (recipient) => this.mozMobileMessageClient.method(
```

Comment:
optional nit: up to you, but I'd rather use outer ```then(successCallback, errorCallback)``` instead of inner ```then(successCallback)``` and outer ```catch(errorCallback)```.